datastore/writer/flask_frontend/routes_handler.py

```diff
@@ -37,6 +41,24 @@ def reserve_ids():
     return JsonResponse({"ids": ids})
 
 
+@handle_internal_errors
+def write_action_worker():
+    if not request.is_json:
+        raise InvalidRequest("Data must be json")
+    if len(cast(Dict[str, Any], request.json).get("events", ())) != 1:
```

Comment:
maybe extract `cast(Dict[str, Any], request.json)` and `cast(Dict[str, Any], request.json)["events"][0]` into their own variables for better readability. Mayne chained index operators, also mixed with functions, are always a hassle to read ;)

Reply:
done